### DISCUSSION TITLE
maybe then this solution will be faster?

### DISCUSSION THREAD

**Discussion by kravchaCS** [KravchaDev/leetcode#4] [Q&A] [answered]
```
func isPalindrome(x int) bool {
    if x < 0 {
        return false
    } else if x <= 9 {
        return true
    } else if x % 10 == 0 {
        return false
    }
    
    var y int
    for x > y {
        r := x % 10
        x = x / 10
        y = y * 10 + r
        
        if x == y || x / 10 == y {
            return true
        }
    }
    return false
}
```

**Comment by KravchaDev** [comment]
no, it won't be faster either: 50 ms

**Comment by KravchaDev** [comment, marked as answer]
> ```
> func isPalindrome(x int) bool {
>     if x < 0 {
>         return false
>     } else if x <= 9 {
>         return true
>     } else if x % 10 == 0 {
>         return false
>     }
>     
>     var y int
>     for x > y {
>         r := x % 10
>         x = x / 10
>         y = y * 10 + r
>         
>         if x == y || x / 10 == y {
>             return true
>         }
>     }
>     return false
> }
> ```

no, it won't be faster either: 50 ms
